infra/feast-operator/internal/controller/featurestore_controller.go

```diff
@@ -78,10 +77,14 @@ func (r *FeatureStoreReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	currentStatus := cr.Status.DeepCopy()
 
 	// initial status defaults must occur before feast deployment
-	applyDefaultsToStatus(cr)
+	if err := services.ApplyDefaultsToStatus(cr); err != nil {
+		logger.Error(err, "Error updating the FeatureStore status")
+		result = ctrl.Result{Requeue: true, RequeueAfter: RequeueDelayError}
+		return result, err
+	}
```

Comment:
let's revert this change and, instead, add the `IsValidOfflineStoreFilePersistenceType` check to `feast.Deploy()` function... which already returns an error to `deployFeast(ctx, cr)` below. We want to keep the defaults application function simple... no need to get into error handling there.

Reply:
done, and also removed negative test case in `repo_config_test.go` because, I don't want to test that RepoConfig has  `offline_store.type==invalid` when the CRD has `offline.persistence.file.type==invalid`